src/components/general/Alert.js

```diff
@@ -0,0 +1,207 @@
+
+		return (
+			<ScrollView style = { [{ marginVertical: "3%" }, padding] }>
+				<Text style = { text }>{ this.state.message }</Text>
+			</ScrollView>
+		);
+	}
+
+	renderButton() {
+		const { center, text, buttonText, buttonContainer, confirmButton, cancelButton } = styles;
+		const { submit, cancel, type } = this.state;
+
+		let submitTitle = submit && submit.title;
+		let cancelTitle = cancel && cancel.title;
+
+		if (type == "confirmation") {
+			return (
+				<View style = { buttonContainer } >
+					<TouchableOpacity style = { [center, confirmButton] } onPress = { () => this.close("submit") }>
+						<Text style = { [text, buttonText] }>{ submitTitle || "Confirm" }</Text>
+					</TouchableOpacity>
+					<TouchableOpacity style = { [center, cancelButton] } onPress = { () => this.close("cancel") }>
+						<Text style = { [text, buttonText] }>{ cancelTitle || "Cancel" }</Text>
+					</TouchableOpacity>
+				</View>
+			);
+		}
+		else {
+			return (
+				<TouchableOpacity style = { [center, buttonContainer] } onPress = { () => this.close("submit") }>
+					<Text style = { [text, buttonText] }>{ submitTitle || "Ok" }</Text>
+				</TouchableOpacity>
+			);
+		}
+	}
+
+	render() {
+		const { modalContainer, alertContainer } = styles;
+
+		return (
+			<View>
+				<Modal transparent = { true } visible = { this.state.visible }>
+					<View style = { modalContainer }>
+						<Animate.Animate
+							type = "View"
+							property = "marginTop"
+							containerStyle = { alertContainer }
+							initial = {{ ratio: 1, screen: "height" }}
+						>
+							{ this.renderTitle() }
+							{ this.renderMessage() }
+							{ this.renderButton() }
+						</Animate.Animate>
+					</View>
+				</Modal>
+			</View>
+		);
+	}
+}
+
+/**
+ * @classdesc Utility class for the AlertComponent. Used to display the alert dialog with the static alert() function,
+ *            similar to the native alert box.
+*/
+
+export class Alert {
+	static AlertBox = () => <AlertComponent ref = { alert => (this.AlertComponent = alert) } />;
+
+	/**
+	 * @param {string}   message               Message for alert to display.
+	 * @param {Object}   config
+	 * @param {string=}  config.title          Title of alert.
+	 * @param {string=}  config.type           Type of alert to display.
+	 *                                         ["alert", "confirmation", "error", "success"]
+	 * @param {Object}   config.submit         Config or function to execute with submit button.
+	 * @param {string}   config.submit.title   Title of confirm button.
+	 * @param {Function} config.submit.onPress onPress of confirm button.
+	 * @param {Object}   config.cancel         Config or function to execute with cancel button.
+	 * @param {string}   config.cancel.title   Title of cancel button.
+	 * @param {Function} config.cancel.onPress onPress of cancel button.
+ */
+	static alert(message, { title = "Alert", type = "alert", submit, cancel } = { title: "Alert", type: "alert" }) {
```

Comment:
Why do you you have { title: "Alert", type: "alert" } twice

Reply:
Also I don't like the idea of that being de-referenced here. You can set those as the default values without de-referencing that in the command line.

Reply:
`alert(message, config= { title: "Alert", type: "alert" })`